jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java

```diff
@@ -54,17 +54,28 @@
     public HonoReceiver(final HonoReceiverSampler sampler) throws InterruptedException {
         this.sampler = sampler;
 
-        // amqp network config
-        amqpNetworkConnectionFactory = ConnectionFactoryImpl.ConnectionFactoryBuilder.newBuilder()
-                .disableHostnameVerification()
-                .host(sampler.getHost())
-                .name(sampler.getContainer())
-                .user(sampler.getUser())
-                .password(sampler.getPwd())
-                .port(Integer.parseInt(sampler.getPort()))
-                .trustStorePath(sampler.getTrustStorePath())
-                .vertx(vertx)
-                .build();
+        if (sampler.getUser() != null && sampler.getPwd() != null) {
```

Comment:
it is hard to see why this should make any difference since the `ConnectionFactoryImpl` will try to retrieve the username and password from its config objetc using `ClientConfigProperties.getUsername()` and `ClientConfigProperties.getPassword()` respectively which return `null`, if no username/password is set.

Reply:
Exactly after this ```null``` is set in the builder, which leads to the error mentioned above in this PR description.

```Could not find a suitable SASL mechanism for the remote peer using the available credentials```

Reply:
my point is that I cannot find any code that your change would have an impact on. Can you point me to any such place?